src/array.jl

```diff
@@ -463,6 +463,21 @@ function compact{T, N}(A::CatArray{T, N})
     convert(arraytype(typeof(A)){T, N, R}, A)
 end
 
+"""
+    uncompact(A::CategoricalArray)
+    uncompact(A::NullableCategoricalArray)
+
+Return a copy of categorical array `A` using the default reference type. If `A` is using
+a small reference type (such as UInt8 or UInt16) the uncompact array will have room for
+more levels.
+
+Avoid using compact to avoid having to call uncompact.
```

Comment:
"compact" should be a link, see how `@ref` is used elsewhere. Though I find this phrasing a bit weird. Maybe something along the lines of "To avoid the need to call `uncompact`, ensure `compact` is not called during creation or importation of data as categorical arrays."?